Makefile.all

```diff
@@ -3,57 +3,67 @@
 # This file defines all packages available for building and defines their
 # dependencies, so that make can build them in parallel effectively.
 #
+# Each package that can be built by conan contains a `conanfile.py`
+# and a Makefile that includes `Makefile.package`.
+# By default all plugin packages are built.
+#
+# Use WITHOUT_PKGS variable on the command line to opt-out of specific
+# packages:
+#
+#     make WITHOUT_PKGS="plugins/demo_printer" package
+#
+# This extends the UNSELECT_PKGS variable. Set this variable directly
+# to override default omitted packages. Use make help to see which
+# these are. For example, if you want to export all vendor packages:
+#
+#	   make UNSELECT_VENDOR= export-vendor
+#
+# Use WITH_PKGS variable on the command line to include specific by-default
+# unselected packages:
+#
+#     make PLUGIN_PKGS= WITH_PKGS="plugins/minimator" package
+#
+# This extends the SELECT_PKGS variable. Set this variable directly
+# to explicitely specify exactly which packages are considered. Use make help
+# to see what these are.
+#
+
+include Makefile.help
 
 # Make configuration:
 SHELL := /bin/bash
 GNUMAKEFLAGS := --no-print-directory
 SUBMAKEFLAGS :=
 
-define assert_numeric_bool
-$(if $(filter-out 0 1,${$1}), $(error Unexpected value for parameter $1, expect either 0 or 1))
-endef
+META_PKG := cloe
+PLUGIN_PKGS := $(wildcard plugins/*)
+ALL_PKGS := fable runtime models oak engine ${PLUGIN_PKGS} ${META_PKG}
+WITHOUT_PKGS :=
+UNSELECT_PKGS := ${WITHOUT_PKGS}
+WITH_PKGS :=
+SELECT_PKGS := $(call uniq, $(filter-out ${UNSELECT_PKGS}, ${ALL_PKGS}) ${WITH_PKGS})
+.PHONY: ${ALL_PKGS}
 
-## USE_NPROC
-##   Usage: make USE_NPROC=1 <targets>
-##   Default: 0
+## VENDOR PACKAGE SELECTION
 ##
-##   If set to 1, this will force $(nproc) number of package builds at a
-##   time, taking dependencies into account. This provides up to a ~50%
-##   speed-up, but makes reading error messages more difficult.
+## Functions analogously to normal package selection.
 ##
-##   You can override the number of jobs by also setting NPROC. For example:
-##
-##       make USE_NPROC=1 NPROC=3 <targets>
-##
-USE_NPROC := 0
-$(call assert_numeric_bool, USE_NPROC)
-
-ifeq (${USE_NPROC},1)
-NPROC := $(shell nproc 2>/dev/null)
-ifdef NPROC
-SUBMAKEFLAGS += -j${NPROC}
-endif
-endif
+ALL_VENDOR := $(wildcard vendor/*)
+WITHOUT_VENDOR :=
```

Comment:
```suggestion
WITHOUT_VENDOR := $(wildcard optional/*/vendor/*)
```

Reply:
Ok, after a bit of trial and error it seems to me that the vendor packages of optional plugins are not meant to be included using `WITH_VENDOR`...
If it was intended, L51 would need to be changed as well:
`ALL_VENDOR := $(wildcard vendor/*) $(wildcard optional/*/vendor/*)`

(what I tried is this:
```
make export-vendor  \ 
WITH_VENDOR="optional/vtd/vendor/osi-sensor-1.0.0-vtd2.2  optional/vtd/vendor/vtd-api  optional/vtd/vendor/protobuf-2.6.1  optional/vtd/vendor/open-simulation-interface-3.0.1  optional/vtd/vendor/open-simulation-interface-3.2.0" \
WITHOUT_VENDOR="vendor/incbin vendor/cpp-netlib vendor/libbacktrace"
```
)

Reply:
Yes, you are right. It was never to be included at this stage.
The correct method is simply to go into the relevant optional directories and build them from there.